### PR TITLE
AutoWarmth: add a time offset for toggling frontlight on/off

### PR DESCRIPTION
**Pull request by zwim**
On cloudy autumn-evenings the light might be too dark to read without frontlight.
This PR adds an offset to the toggling time, so that the fl might be toggled some time before sunset or after sunrise.

Wordings are welcomed.

Draft by now, as this will need some more testing. (Easy-mode, expert-mode, user fl toggle, sleep and resume ...)

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9755)

### CODE REVIEW

File: plugins/autowarmth.koplugin/main.lua [hunk 1]
```diff
@@ -581,38 +604,86 @@ function AutoWarmth:getSubMenuItems()
...
+        end,
+        callback = function(touchmenu_instance)
+            if self.easy_mode then
                 self.fl_off_during_day = not self.fl_off_during_day
                 G_reader_settings:saveSetting("autowarmth_fl_off_during_day", self.fl_off_during_day)
+            else
+                -- hard limit offset to 15 min after sunset and 30 mins before sunset; sunrise vice versa
+                UIManager:show(SpinWidget:new{
+                    title_text = _("Offset for frontlight off"),
+                    info_text = _[[This is the time before sunset (after sunrise), when the frontlight turns off.
```
Comment (Frenzie) [inline plugins/autowarmth.koplugin/main.lua:634]: Is this saying "This is the first time… that the frontlight has turned off"?
```suggestion
                    info_text = _[[This is the time before sunset (after sunrise), when the frontlight turns off.
```

Comment (zwim) [reply plugins/autowarmth.koplugin/main.lua:634]: I want to say: Frontlight gets turned off from sunsrise to sunset. With this setting the off-time can be from sunrise plus offset to sunset minus offset.

Comment (Frenzie) [reply plugins/autowarmth.koplugin/main.lua:634]: Like this?
```suggestion
                    info_text = _[[The frontlight will be turned off at this time before sunset (after sunrise).
```

**Comment by zwim** [comment]
Ready to merge?

**Review by Frenzie — APPROVED** [review]
If it's ready to you. :+1:

**Comment by zwim** [comment]
Ready.

an optimized version with improved schedulings and other  (mini);optimizations will follow for 2022.12

**Pull request merged**